File: benchmark/corpus/python/xss_vuln_2.py
"""VULN: CWE-79 — Reflected XSS via unescaped user input in Flask response."""
import flask

app = flask.Flask(__name__)


@app.route("/search")
def search():
    query = flask.request.args.get("q", "")
    html = "<html><body><h2>Results for: " + query + "</h2></body></html>"
    return flask.Response(html, content_type="text/html")

File: benchmark/corpus/javascript/xss_vuln_2.js
// VULN: CWE-79 — Reflected XSS via res.send() with unescaped user input
const express = require('express');
const app = express();

app.get('/search', (req, res) => {
    const query = req.query.q;
    res.send('<html><body><h2>Results for: ' + query + '</h2></body></html>');
});

app.listen(3000);
